packages/xstate-react/src/createActorContext.ts

```diff
@@ -93,16 +108,28 @@ export function createActorContext<TMachine extends AnyStateMachine>(
   }
 
   function useSelector<T>(
-    selector: (snapshot: SnapshotFrom<TMachine>) => T,
+    selector: (snapshot: SnapshotFrom<TBehavior>) => T,
     compare?: (a: T, b: T) => boolean
   ): T {
     const actor = useContext();
     return useSelectorUnbound(actor, selector, compare);
   }
 
+  function useSnapshot(): [
+    SnapshotFrom<TBehavior>,
+    (event: EventFromBehavior<TBehavior>) => void,
+    ActorRefFrom<TBehavior>
+  ] {
+    const actor = useContext();
+    const state = useSelector((s) => s);
+
+    return [state, (actor as AnyActorRef).send, actor];
```

Comment:
Yeah, same argument can be made against `useState` though

Reply:
hm, yee - but it creates and own the state so it makes slightly more sense there that they have to give us the setter *somehow* (through the return value). This code only consumes the `actorRef` and the "setter" (`send`) already exists... but ye, since we don't have any alternative ideas here... this seems to be good enough.